components/Sidebar.tsx

```diff
@@ -15,17 +15,36 @@ export const Sidebar = () => {
   const { languages, tags } = useAppData();
   const { tag: activeTagId, language: activeLanguageId } = router.query;
   const pageName = router.pathname.split("/")[1];
+  const [scrollHeightReached, setScrollHeightReached] = useState(false);
 
   useEffect(() => {
     const scrollTarget = document.getElementById("repositories-list");
     const isMobile = window.innerWidth <= 640;
     if (isMobile && (pageName === "language" || pageName === "tag")) {
       scrollTarget?.scrollIntoView({ behavior: "smooth" });
     }
+
+    // Function to handle the scroll event
+    const shouldApplyStickyClasses = () => {
+      // Check the scroll position to determine when to apply classes
+      if (window.scrollY >= 701.5) {
+        setScrollHeightReached(true);
+      } else {
+        setScrollHeightReached(false);
+      }
+    };
+
+    // Add the scroll event listener
+    window.addEventListener("scroll", shouldApplyStickyClasses);
```

Comment:
We have similar code in the scroll component itself, can they be unified somehow?